Sources/AppStoreConnectClient/AppStoreConnectClient.swift

```diff
@@ -55,45 +55,73 @@ public struct AppStoreConnectClient {
         let response = try await client.apps_hyphen_get_collection()
         switch response {
         case .ok(let okResponse):
-            switch okResponse.body {
-            case .json(let json):
-                return json.data.compactMap({ Application(schema: $0) })
-            }
+            return try okResponse.body.json.data.compactMap({ Application(schema: $0) })
         case .badRequest(let result):
-            throw AppStoreConnectError.badRequest
+            throw AppStoreConnectError.badRequest(
+                errors: AppStoreConnectError.errorDescription(from: try result.body.json)
+            )
         case .forbidden(let result):
-            throw AppStoreConnectError.forbidden
+            throw AppStoreConnectError.forbidden(
+                errors: AppStoreConnectError.errorDescription(from: try result.body.json)
+            )
         case .unauthorized(let result):
-            throw AppStoreConnectError.unauthorized
+            throw AppStoreConnectError.unauthorized(
+                errors: AppStoreConnectError.errorDescription(from: try result.body.json)
+            )
         case .undocumented(let statusCode, _):
-            throw AppStoreConnectError.serverError(errorCode: statusCode)
+            throw AppStoreConnectError.serverError(
+                errorCode: statusCode
+            )
         }
     }
     
     /// Fetches a collection of versions for a specified app from the App Store Connect API.
     /// - Parameter app: The app for which to fetch versions.
     /// - Returns: An array of `Release` objects.
-    /// - Throws: An error if the fetch operation fails.
+    /// - Throws: An error of type `AppStoreConnectError` if the fetch operation fails.
     public func fetchVersions(for app: Application) async throws -> [Release] {
         let response = try await client.apps_hyphen_appStoreVersions_hyphen_get_to_many_related(
             path: .init(id: app.id)
         )
         switch response {
         case .ok(let okResponse):
-            switch okResponse.body {
-            case .json(let json):
+            if case .json(let json) = okResponse.body {
                 return json.data.compactMap({ Release(schema: $0) })
             }
+        default:
+            return try errorResponse(from: response)
+        }
+        throw AppStoreConnectError.unexpectedError(errors: "\(response)")
+    }
+    
+    /// Handles error responses returned by the App Store Connect API.
+    /// - Parameter response: The response received from the API.
+    /// - Throws: An error of type `AppStoreConnectError` if the response indicates an error.
+    private func errorResponse(
+        from response: Operations.apps_hyphen_appStoreVersions_hyphen_get_to_many_related.Output
+    ) throws -> [Release] {
```

Comment:
do we need to return anything from this func?

Reply:
Deleted "return".

Reply:
this function does not return anything, so you need to remove what it returns - [release]